packages/sync-actions/src/utils/create-build-array-actions.js

```diff
@@ -23,26 +23,42 @@ export default function createBuildArrayActions (key, config) {
 
     if (diff[key]) {
       const arrayDelta = diff[key]
+
       Object.keys(arrayDelta).forEach((index) => {
-        if (config[ADD_ACTIONS] && isCreateAction(arrayDelta, index))
-          addActions.push(
-            // When adding a new element you don't need the oldObj
-            config[ADD_ACTIONS](newObj[key][index]),
+        if (config[ADD_ACTIONS] && isCreateAction(arrayDelta, index)) {
+          const actionGenerator = config[ADD_ACTIONS]
```

Comment:
Just general simplications in naming things here. The addition is to pass the index to the callback so it has the information.